charm4py/charmlib/charmlib_cffi_build.py

```diff
@@ -6,8 +6,10 @@
 
 if platform.system() == 'Darwin':
     extra_link_args=["-Wl,-rpath,@loader_path/../.libs"]
+    extra_compile_args=['-g0', '-O3', '-Wno-error=incompatible-function-pointer-types']
```

Comment:
Is adding g0/O3 necessary for the build?

Reply:
Those look like they come from the initial development of the build (#11). I'm not sure what you mean by necessary, since they seem like they're just optimizations? When I build this locally, -O3 doesn't do anything but -g0 shrinks the executable size substantially. Should I look into this more?

Reply:
Ohh okay makes sense. So we'd want to remove them from setup.py as well? For the user to have to add these back in though, I don't see any way to do that other than editing setup.py/charmlib_cffi_build.py directly?

Reply:
There may be a way to pass-through `CFLAGS` or so, I think. I'd suggest removing those options from setup.py too. With those options removed, what are the optimization options you get on your system?

Reply:
On my system, clang has default flags `g` and `03`. The cffi command looks like this after removing the optimization flags from the charm4py source:

`
clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -Wno-error=implicit-function-declaration -arch arm64 -Icharm_src/charm/include -I/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c build/temp.macosx-14.0-arm64-cpython-311/charm4py.charmlib._charmlib_cffi.c -o build/temp.macosx-14.0-arm64-cpython-311/build/temp.macosx-14.0-arm64-cpython-311/charm4py.charmlib._charmlib_cffi-both.o -Wno-error=incompatible-function-pointer-types
`

Exporting CXXFLAGS/CFLAGS doesn't seem to be enough to add the original flags back in.

Reply:
Edit: setting CPPFLAGS or CFLAGS allow you to add the original flags back in.

Reply:
Ok, this looks good. I’d suggest removing the opt flags here.